### PR TITLE
Upstream changes for v0.8.0 release

### PR DESCRIPTION
**Pull request by shubhadeepd**
This release completely refactors the directory structure of the repository for a more seamless and intuitive developer journey. It also adds support to deploy latest accelerated embedding and reranking models across the cloud, data center, and workstation using [Nvidia NeMo Retriever NIMs](https://docs.nvidia.com/nim/index.html#nemo-retriever).

For more details refer to the changes in `CHANGELOG.md` file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+- [End to end RAG examples](./RAG/examples/) enhancements
+  - [Single command deployment](./README.md#try-it-now) for all the examples using docker compose.
+  - All end to end RAG examples are now more encapsulated with documentation, code and deployment assets residing in dedicated example specific directory.
+  - Segregated examples into [basic and advanced RAG](./RAG/examples/) with dedicated README's.
+  - Added reranker model support to [multi-turn RAG example](./RAG/examples/advanced_rag/multi_turn_rag/)
+  - Removed python dev packages from containers to enhance security
+  - Updated to latest version of [langchain-nvidia-ai-endpoints](https://python.langchain.com/v0.2/docs/integrations/providers/nvidia/)
+- [Speech support using rag-playground]((./docs/riva-asr-tts.md))
+  - Added support to access [RIVA speech models from NVIDIA API Catalog](https://build.nvidia.com/explore/speech).
+  - Speech support in UI is turned off as default.
```
Comment (dglogo) [inline CHANGELOG.md:21]: Can we please rephrase "tuned off by default" to "opt-in"? Thanks

**Review by dglogo — APPROVED** [review]
Thanks for putting this together @shubhadeepd! Add a small nit, but otherwise lgtm! 
Thanks

File: CHANGELOG.md (hunk 1 above)
Comment (shubhadeepd) [reply CHANGELOG.md]: Changed!

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 
+## [0.8.0] - 2024-08-19
+
+This release completely refactors the directory structure of the repository for a more seamless and intuitive developer journey. It also adds support to deploy latest accelerated embedding and reranking models across the cloud, data center, and workstation using [Nvidia NeMo Retriever NIMs](https://docs.nvidia.com/nim/index.html#nemo-retriever).
```
Comment (mikemckiernan) [inline CHANGELOG.md:9]: Branding is murder: [NVIDIA NeMo Retriever NIM microservices](https:/...)

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+
+### Added
+- [End to end RAG examples](./RAG/examples/) enhancements
```
Comment (mikemckiernan) [inline CHANGELOG.md:12]: nit: End-to-end

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+  - [Single command deployment](./README.md#try-it-now) for all the examples using docker compose.
```
Comment (mikemckiernan) [inline CHANGELOG.md:13]: nits: Single-command deployment and Docker Compose.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+  - All end to end RAG examples are now more encapsulated with documentation, code and deployment assets residing in dedicated example specific directory.
+  - Segregated examples into [basic and advanced RAG](./RAG/examples/) with dedicated READMEs.
+  - Added reranker model support to [multi-turn RAG example](./RAG/examples/advanced_rag/multi_turn_rag/)
+  - Removed python dev packages from containers to enhance security
```
Comment (mikemckiernan) [inline CHANGELOG.md:17]: nit: Python

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+  - Updated to latest version of [langchain-nvidia-ai-endpoints](https://python.langchain.com/v0.2/docs/integrations/providers/nvidia/)
+- [Speech support using rag-playground]((./docs/riva-asr-tts.md))
+  - Added support to access [RIVA speech models from NVIDIA API Catalog](https://build.nvidia.com/explore/speech).
+  - Speech support in UI is opt-in.
```
Comment (mikemckiernan) [inline CHANGELOG.md:21]: My $0.02 is to use "RAG Playground" and treat it like a proper noun--that's what we do for products and projects.

s/UI/RAG Playground/, please.

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+- Documentation enhancements
+  - Added more comprehensive [how-to guides](./README.md#how-to-guides) for end to end RAG examples.
+  - Added [example specific architecture diagrams](./RAG/examples/basic_rag/langchain/) in each example directory.
+- Added a new industry specific [top level directory](./industries/)
+  - Added [health care domain specific Medical Device Training Assistant RAG](./industries/healthcare/medical-device-training-assistant/).
+- Added notebooks showcasing new usecases
+  - [Basic langchain based RAG pipeline](./RAG/notebooks/langchain/langchain_basic_RAG.ipynb) using latest Nvidia API Catalog connectors.
+  - [Basic llamaindex based RAG pipeline](./RAG/notebooks/llamaindex/llamaindex_basic_RAG.ipynb) using latest Nvidia API Catalog connectors.
```
Comment (mikemckiernan) [inline CHANGELOG.md:29]: s/Nvidia/NVIDIA/g

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+  - [NeMo Guardrails with basic langchain RAG](./RAG/notebooks/langchain/NeMo_Guardrails_with_LangChain_RAG/).
+  - [NVIDIA NIMs using NeMo Guardrails based RAG](./RAG/notebooks/langchain/Using_NVIDIA_NIMs_with_NeMo_Guardrails/).
```
Comment (mikemckiernan) [inline CHANGELOG.md:31]: s/NIMs/NIM microservices/

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+  - [Using NeMo Evaluator using Llama 3.1 8B Instruct](./RAG/notebooks/nemo/Nemo%20Evaluator%20Llama%203.1%20Workbook/).
+  - [Agentic RAG pipeline with Nemo Retriever and NIM for LLMs](./RAG/notebooks/langchain/agentic_rag_with_nemo_retriever_nim.ipynb).
+- Added new `community` (before `experimental`) example
+  - Create a simple UI to interact with different [selectable NIM endpoints](./community/llm-prompt-design-helper/). The provided interface of this project supports designing a system prompt to call the LLM.
```
Comment (mikemckiernan) [inline CHANGELOG.md:35]: nit: s/UI/web interface/

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,58 @@ All notable changes to this project will be documented in this file.
...
+
+### Changed
+- Major restructuring and reorganisation of the assets within the repository
+  - Top level `experimental` directory has been renamed as `community`.
+  - Top level `RetrievalAugmentedGeneration` directory has been renamed as just `RAG`.
+  - The docker compose files inside top level `deploy` directory has been migrated to example specific directories under `RAG/examples`. The vector db and on-prem NIMs deployment files are under `RAG/examples/local_deploy`.
```
Comment (mikemckiernan) [inline CHANGELOG.md:41]: Docker Compose, example-specific, vector database, s/NIMs/NIM microservices/

File: CHANGELOG.md (hunk 3 above)
Comment (shubhadeepd) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 5 above)
Comment (shubhadeepd) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 6 above)
Comment (shubhadeepd) [reply CHANGELOG.md]: Replaced

File: CHANGELOG.md (hunk 9 above)
Comment (shubhadeepd) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md (hunk 10 above)
Comment (shubhadeepd) [reply CHANGELOG.md]: Thanks made all the changes

**Review by mikemckiernan — DISMISSED** [review]
Thanks very much!

**Review by dglogo — APPROVED** [review]
Thank you @shubhadeepd !

**Review by sumitkbh — APPROVED** [review]

**Review by dglogo — APPROVED** [review]

**Review by Trile9502 — COMMENTED** [review]
Personalized Character
Demo Overview
This workflow leverages Generative AI to turn any person into an illustrated character. It chains together 5 distinct AI workflows to produce character sheets and an animated turntable video. On an NVIDIA GPU, running on a powerful workstation, all of this content can be composed in just under 1.5 minutes.

Finally everything comes together though compositing automation and SAM . Customized background elements, text and generated assets come together to create the final composition.

Key Points
Generate a specific character in a T-pose, based off of real life images - Using Instant ID & IP Adapter Face, Controlnet Open Pose and Stable Diffusion XL \

Generate a full body character in multiple angles using Diffusers based Multi View

Upscale Illustrations and create high quality faces with Generative AI upscale methods, Face Detailer and Face Detection Ultralytics , and SD Ultimate Upscale.

Create unique character expressions for your character sheet using Live Portrait

Extract Key colors and elements from the scene using AI color extractors

**Pull request merged**